aiohttp/web_fileresponse.py

```diff
@@ -46,54 +53,54 @@ def _write(self, chunk: bytes) -> None:
         self.output_size += len(chunk)
         self._sendfile_buffer.append(chunk)
 
-    def _sendfile_cb(self, fut: 'asyncio.Future[None]',
-                     out_fd: int, in_fd: int,
-                     offset: int, count: int,
-                     loop: asyncio.AbstractEventLoop,
-                     registered: bool) -> None:
-        if registered:
-            loop.remove_writer(out_fd)
+    def _sendfile_cb(self, fut: 'asyncio.Future[None]', out_fd: int) -> None:
         if fut.cancelled():
             return
+        try:
+            if self._do_sendfile(out_fd):
+                set_result(fut, None)
+        except Exception as exc:
+            set_exception(fut, exc)
 
+    def _do_sendfile(self, out_fd: int) -> bool:
         try:
-            n = os.sendfile(out_fd, in_fd, offset, count)
+            n = os.sendfile(out_fd,
+                            self._in_fd,
+                            self._offset,
+                            self._count)
             if n == 0:  # EOF reached
-                n = count
+                n = self._count
         except (BlockingIOError, InterruptedError):
```

Comment:
Seems Interrupted error EINTR ? may not happen since Python 3.5 (I'm unsure)

Reply:
It was in the previous version, I'd like to keep it now.